src/coreclr/src/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -3957,112 +3951,196 @@ private void CreateInstanceCheckThis()
+                        _pfnCtorStub = &ReferenceTypeStub;
+                        _ctorStubState = RuntimeMethodHandle.GetFunctionPointer(defaultCtorRMH);
+                    }
+                }
+
+                if (defaultCtorRMH.IsNullHandle())
+                {
+                    CtorIsPublic = true; // implicit parameterless ctor is always considered public
+                }
+                else
+                {
+                    CtorIsPublic = (RuntimeMethodHandle.GetAttributes(defaultCtorRMH) & MethodAttributes.Public) != 0;
+                }
+
+                Debug.Assert(_pfnNewobj != null);
+                Debug.Assert(_pfnCtorStub != null); // we use null singleton pattern if no ctor call is necessary
             }
 
-            if (canBeCached && fillCache)
+            internal bool CtorIsPublic { get; }
+
+            private static void CtorNoopStub(object? @this, IntPtr state) { }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            internal object? CreateUninitializedObject(RuntimeType rt)
             {
-                // cache the ctor
-                GenericCache = new ActivatorCache(runtimeCtor);
+                // We don't use RuntimeType, but we force the caller to pass it so
+                // that we can keep it alive on their behalf. Once the object is
+                // constructed, we no longer need the reference to the type instance,
+                // as the object itself will keep the type alive.
+
+#if DEBUG
+                Debug.Assert(rt == _originalRT, "Caller passed the wrong RuntimeType to this routine.");
+#endif
+
+                object? retVal = _pfnNewobj(_pMT);
+                GC.KeepAlive(rt);
+                return retVal;
             }
 
-            return instance;
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            internal void CallCtorOverUninitializedObject(object? newObj) => _pfnCtorStub(newObj, _ctorStubState);
+        }
+
+        // This method mimics the overload present in the mono codebase. It allows shared source callers
+        // to target this overload and work across both runtimes. CoreCLR ignores the 'skipCheckThis' and
+        // 'fillCache' parameters.
+        [DebuggerStepThrough]
+        [DebuggerHidden]
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2082:UnrecognizedReflectionPattern",
+            Justification = "Implementation detail of Activator that linker intrinsically recognizes")]
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2085:UnrecognizedReflectionPattern",
+            Justification = "Implementation detail of Activator that linker intrinsically recognizes")]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal object? CreateInstanceDefaultCtor(bool publicOnly, bool skipCheckThis, bool fillCache, bool wrapExceptions)
+        {
+            return CreateInstanceDefaultCtor(publicOnly, wrapExceptions);
```

Comment:
What does `AggressiveInlining` do for this one-liner? (and elsewhere for one-liners).